spec/fuzzy-finder-spec.js

```diff
@@ -254,13 +254,13 @@ describe('FuzzyFinder', () => {
         describe('socket files on #darwin or #linux', () => {
           let socketServer, socketPath
 
-          beforeEach(() => {
+          beforeEach(async () => {
             socketServer = net.createServer(() => {})
             socketPath = path.join(rootDir1, 'some.sock')
-            waitsFor(done => socketServer.listen(socketPath, done))
+            await socketServer.listen(socketPath)
```

Comment:
Looking at [the docs for `Server.listen`](https://nodejs.org/api/net.html#net_server_listen), it looks like the method just returns the `Server` itself, and not a `Promise`. So I would think that using `await` on the return value would be a noop.

It doesn't look like [`close`](https://nodejs.org/api/net.html#net_server_close_callback) returns a promise either, so `await` should be a noop there too.

I would think that you'd have to pass a callback to both methods; something like:

```js
await new Promise(resolve => socketServer.listen(socketPath, resolve))

// ...

await new Promise(resolve => socketServer.close(resolve))

```

It's interesting that this fixes the tests though. Maybe the listening and closing are so fast that they already complete before the next block executes?